portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,31 +20,42 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    ArrayList<String> messages = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
-    response.setContentType("application/json;");
+    Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
     String json = "[";
```

Comment:
Consider using the StringBuilder class to do appends like these.  It's a lot more efficient than java's string concatenation, which copies the entire string every time you concatenate.

Reply:
Also - surely there's some kinda good JSON class in Java (though I don't know for sure what it is).  I feel like there ought to be a class where you can just take an ArrayList (or any Java list, really) and do something like Json.Convert() to it to get the JSON that represents the list of items.  Perhaps ask the tech support folks what they recommend?

Reply:
Ok I'll check it out